registration-processor/core-processor/registration-processor-reprocessor-stage/src/main/java/io/mosip/registration/processor/reprocessor/service/WorkflowSearchService.java

```diff
@@ -0,0 +1,137 @@
+import io.mosip.registration.processor.status.dto.RegistrationStatusDto;
+import io.mosip.registration.processor.status.service.RegistrationStatusService;
+
+@Component
+public class WorkflowSearchService {
+
+	/** The registration status service. */
+	@Autowired
+	RegistrationStatusService<String, InternalRegistrationStatusDto, RegistrationStatusDto> registrationStatusService;
+
+	/** The core audit request builder. */
+	@Autowired
+	AuditLogRequestBuilder auditLogRequestBuilder;
+
+	/** The module name. */
+	public static String MODULE_NAME = ModuleName.WORKFLOW_ACTION_SERVICE.toString();
+
+	/** The module id. */
+	public static String MODULE_ID = PlatformSuccessMessages.RPR_WORKFLOW_ACTION_SERVICE_SUCCESS.getCode();
+
+	/** The reg proc logger. */
+	private static Logger regProcLogger = RegProcessorLogger.getLogger(WorkflowActionService.class);
+
+	public PageResponseDto<SearchResponseDto> searchRegistrationDetails(SearchDto searchDto)
+			throws WorkFlowSearchException {
+
+		PageResponseDto<SearchResponseDto> pageResponseDto = new PageResponseDto<>();
+		List<SearchResponseDto> searchResDtos = new ArrayList<SearchResponseDto>();
+		regProcLogger.debug(LoggerFileConstant.SESSIONID.toString(), LoggerFileConstant.USERID.toString(), "",
+				"WorkflowSearchService::searchRegistrationDetails()::entry");
+		try {
+			Page<InternalRegistrationStatusDto> pageDtos = registrationStatusService.searchRegistrationDetails(searchDto);
+
+			regProcLogger.debug(LoggerFileConstant.SESSIONID.toString(), LoggerFileConstant.USERID.toString(), "",
+					"WorkflowSearchService::searchRegistrationDetails()::exit");
+
+			if (!pageDtos.getContent().isEmpty()) {
+				for (InternalRegistrationStatusDto regs : pageDtos.getContent()) {
+					SearchResponseDto searchRes = dtoMapper(regs);
+					searchResDtos.add(searchRes);
+				}
+
+			}
+			pageResponseDto = pageResponse(pageDtos);
+			pageResponseDto.setData(searchResDtos);
+
+		} catch (DataAccessLayerException e) {
+			regProcLogger.error(LoggerFileConstant.SESSIONID.toString(), LoggerFileConstant.REGISTRATIONID.toString(),
+					"", e.getMessage() + ExceptionUtils.getStackTrace(e));
+			logAndThrowError(e, PlatformErrorMessages.RPR_RGS_REGISTRATION_TABLE_NOT_ACCESSIBLE.getCode(),
+					PlatformErrorMessages.RPR_RGS_REGISTRATION_TABLE_NOT_ACCESSIBLE.getMessage(), null);
+		}
+		return pageResponseDto;
+
+	}
+
+	public static <T, D> PageResponseDto<D> pageResponse(Page<T> page) {
+		PageResponseDto<D> pageResponse = new PageResponseDto<>();
+		if (page != null) {
+			long totalItem = page.getTotalElements();
+			int pageSize = page.getSize();
+			int start = (page.getNumber() * pageSize) + 1;
+			pageResponse.setFromRecord(start);
+			pageResponse.setToRecord((long) (start - 1) + page.getNumberOfElements());
+			pageResponse.setTotalRecord(totalItem);
+		}
+		return pageResponse;
+	}
+
+	private SearchResponseDto dtoMapper(InternalRegistrationStatusDto regSt) {
```

Comment:
Method should be named as verb, so rename this also to convertToWorkflowDetail

Reply:
taken care